### ISSUE TITLE
is there any option for the pokemon with the low CP to be removed even if they have the minimum IV?

### ISSUE THREAD

**Issue by DeOrthodox** [j-e-k/poketrainer#126]
I dont know if this is the right place to ask but i cant find anyother place where i could get help. I have a problem where the pokemons have filled up my bot. and its full off 10 CP pokemon and such with 60% IV 
i have 60% IV as the minimum and i have 800CP and the minimum but i still have the 10 CP Zubats with 60+% IV coming through.... is there a way to make the bot remove all the pokemons with lower CP than the minimum even if they have the minimum IV?

**Comment by j-e-k** [comment]
I am worried about the configuration bloat that is occurring with all of these different options. Afraid someone might get confused and end up deleting all of their pokemon..... Give me a bit to decide how to proceed forward in general and I will add this in. Thanks! 😄

**Comment by DeOrthodox** [comment]
sounds like a plan! aslong as it is on the agenda ;D ill just put the IV req for 90% :P

**Comment by Zyrixion** [comment]
One of the newer features is a list of pokemon that simply get released no matter what, which I think would solve your zubat problem (who needs zubats anyways).  As well, 60% IV is too low imo, and that's what the real problem is with your bag filling.  I have zubats/golbats on always release (since my area is buried in them enough to evolve them), and IVs at 80%, and I only fill about 40-60 slots in a day that I have to manually deal with.

**Comment by Zyrixion** [comment]
@j-e-k as a suggestion about implementation, if you do decide that it is necessary (which I'm not sold on), a toggle for the option in the config should suffice, as long as it exempts favorite/keep list pokemon, and has proper documentation in the readme.  it definitely be a toggle, though, and default to off.

Personally I think that just having higher IV standards is enough of a solution to this issue, however.

**Comment by beeedy** [comment]
I have a PR that has been sitting for a while that can be of a lot of use in this type of situation: https://github.com/j-e-k/poketrainer/pull/85

I am working on merging it in with the most recent master/development branches so you may have to insert it yourself into the code. Luckly it is pretty straight forward :)

**Comment by j-e-k** [comment]
Sorry for the delay in getting back to you, I have been really busy lately 😩 .  Can you reopen your pull request #85? Github is being weird with letting me review it (a lot of things have been messed up post name change...).

**Comment by destiny117** [comment]
just fyi, #110 offers the feature you want and was merged and even improved already

you can set "KEEP_IV_MIN_PERCENT_CP" in the config to keep high-ivs only if they are above a % threshold from the strongest and/or limit the amount of pokemons kept for the IV value with "MAX_POKEMON_HIGH_IV"

#141 (previously #81) is a different approach with similar results, though it does not distinguish between pokemons kept for IV and others kept for CP, it uses a custom generated pokemon lvl which needs new thresholds to define what to keep

**Comment by xezero** [comment]
Personally I like #110 over #141 as it more closely resembles what one wants when farming for competitive Pokemon, but for some reason KEEP_IV_MIN_PERCENT_CP is removed from the latest branch in favor of #141.

**Comment by destiny117** [comment]
yes you are right, by merging #141 they completely reverted #110 

i also prefer the method in #110, but i guess it was a bit complicated to understand / set up.
expect a new pull request from me soon to re-enable the functionality of #110

**Comment by beeedy** [comment]
As far as I am aware, none of the functionality to #110 was removed with #141, it was however restructured and the code cleaned up.

**Comment by destiny117** [comment]
nah, unfortunately you made PR #141 shortly after #110 got merged and did not rebase everything, neither did anyone else when merging, so #110 got completely wiped out again

**Comment by beeedy** [comment]
Well apologizes for that, hopefully it isn't much work to slip it back in!

**Comment by destiny117** [comment]
haha np, gj with #141 anyway :)

i see it as an oportunity to make #110 a bit more use friendly

**Comment by destiny117** [comment]
I will close this issue now as this is possible. Specifically you can check the ADVANCED release method option `ALWAYS_RELEASE_BELOW_LEVEL` or the DUPLICATES release method. Check the readme for details.
